### PR TITLE
AuditlogParser update

### LINKED ISSUE #1569: DiscordAuditLogMessageEntry.Target property is of wrong type.
# Summary
The `DiscordAuditLogMessageEntry.Target` property is of type `DiscordMessage`, but should be of type `DiscordUser`

# Details
When parsing the audit log for deleted messages, the `DiscordAuditLogMessageEntry.Target` had a type of `DiscordMessage` but when inspected, the `Target.Id` was that of the User who's message was deleted. Upon reviewing the Discord API Documentation and other sources, I believe the intended behavior is for the `target_id` passed from discord to be that of the User who's message was deleted.

# Notes
I am using version v4.4.1

### PR DESCRIPTION
**Pull request by Plerx2493**
# Summary
- Adds missing keys to channelUpdate auditlog entries
- Fix weird casts
- update PropertyChange
- Performance upgrades
- Auditlogs are now parsed and returned as a IAsyncEnumerable
- Fixes #1569 

# Notes
https://discord.com/channels/379378609942560770/379386901725052928/1147974237735366818

### CODE REVIEW

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 1]
```diff
@@ -980,15 +980,15 @@ public Task UnbanMemberAsync(DiscordUser user, string reason = null)
...
+    /// <param name="userId">ID of the user to unban.</param>
     /// <param name="reason">Reason for audit logs.</param>
     /// <returns></returns>
     /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.BanMembers"/> permission.</exception>
     /// <exception cref="NotFoundException">Thrown when the user does not exist.</exception>
     /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
     /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public Task UnbanMemberAsync(ulong user_id, string reason = null)
-        => this.Discord.ApiClient.RemoveGuildBanAsync(this.Id, user_id, reason);
+    public Task UnbanMember(ulong userId, string reason = null)
```
Comment (akiraveliara) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:990]: this should retain the suffix

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 2]
```diff
@@ -1078,8 +1078,8 @@ public Task<DiscordChannel> CreateChannelCategoryAsync(string name, IEnumerable<
     /// <exception cref="NotFoundException">Thrown when the guild does not exist.</exception>
     /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
     /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public Task<DiscordChannel> CreateVoiceChannelAsync(string name, DiscordChannel parent = null, int? bitrate = null, int? user_limit = null, IEnumerable<DiscordOverwriteBuilder> overwrites = null, VideoQualityMode? qualityMode = null, int? position = null, string reason = null)
-        => this.CreateChannelAsync(name, ChannelType.Voice, parent, Optional.FromNoValue<string>(), bitrate, user_limit, overwrites, null, Optional.FromNoValue<int?>(), qualityMode, position, reason);
+    public Task<DiscordChannel> CreateVoiceChannel(string name, DiscordChannel parent = null, int? bitrate = null, int? userLimit = null, IEnumerable<DiscordOverwriteBuilder> overwrites = null, VideoQualityMode? qualityMode = null, int? position = null, string reason = null)
```
Comment (akiraveliara) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:1081]: this should also retain the suffix

File: DSharpPlus/Entities/Guild/DiscordGuild.cs [hunk 3]
```diff
@@ -1262,16 +1262,16 @@ public Task<DiscordIntegration> AttachUserIntegrationAsync(DiscordIntegration in
...
+    /// <param name="expireGracePeriod">Length of grace period which allows for renewing the integration.</param>
+    /// <param name="enableEmoticons">Whether emotes should be synced from this integration.</param>
     /// <returns>The modified integration.</returns>
     /// <exception cref="UnauthorizedException">Thrown when the client does not have the <see cref="Permissions.ManageGuild"/> permission.</exception>
     /// <exception cref="NotFoundException">Thrown when the guild does not exist.</exception>
     /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
     /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public Task<DiscordIntegration> ModifyIntegrationAsync(DiscordIntegration integration, int expire_behaviour, int expire_grace_period, bool enable_emoticons)
-        => this.Discord.ApiClient.ModifyGuildIntegrationAsync(this.Id, integration.Id, expire_behaviour, expire_grace_period, enable_emoticons);
+    public Task<DiscordIntegration> ModifyIntegration(DiscordIntegration integration, int expireBehaviour, int expireGracePeriod, bool enableEmoticons)
```
Comment (akiraveliara) [inline DSharpPlus/Entities/Guild/DiscordGuild.cs:1273]: as should this

**Review by akiraveliara — CHANGES_REQUESTED** [review]
three naming issues, other than that lgtm

File: DSharpPlus/Entities/Guild/DiscordGuild.cs (hunk 1 above)
Comment (Plerx2493) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:990]: Wasnt intended to be changed

File: DSharpPlus/Entities/Guild/DiscordGuild.cs (hunk 2 above)
Comment (Plerx2493) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:1081]: 2x

File: DSharpPlus/Entities/Guild/DiscordGuild.cs (hunk 3 above)
Comment (Plerx2493) [reply DSharpPlus/Entities/Guild/DiscordGuild.cs:1273]: 3x

**Review by akiraveliara — APPROVED** [review]

**Pull request merged**